src/app/(admin)/new_dashboard/guides/hooks/useSearchGuides.ts

```diff
@@ -12,7 +14,16 @@ export const useSearchGuides = (initialGuides: Guide[]) => {
     const matchesTag =
       selectedTag === "All" || guide.categories.includes(selectedTag);
     return matchesQuery && matchesTag;
-  });
+  })
+    .sort((a, b) => {
+      if (sortCriteria === "rating") {
+        const avgRatingA = a.reviews?.reduce((acc, review) => acc + review.rating, 0) ?? 1;
+        const avgRatingB = b.reviews?.reduce((acc, review) => acc + review.rating, 0) ?? 1;
+        return avgRatingB - avgRatingA;
+      } else {
```

Comment:
how did you decided that "else" nessecirily means by date? why only to hard coded options? what happens if we want to add a new one? make this more abstract and extract buissness logic.
you need to extract all of this logic from this file, and let other functions to handle this case
possible solution :

in this function: `sortByCretiria(sortCriteria )`
and in another file handle the logic of actually sorting

another possible solution is

Reply:
ok